selector.h

```diff
@@ -0,0 +1,145 @@
+
+using namespace std;
+
+class Selector {
+    private:
+        int selected = -1;
+        bool down = false;
+        int up_key_state = 0;
+        int down_key_state = 0;
+        int count = 0;
+
+    public:
+        // Function to check which key is currently pressed and call correct method.
+        // Key states ensure the functions only get called once per key press.
+        void check_key_input(vector<Button*> buttons)
+        {
+            // Down key pressed
+            if (key_down(DOWN_KEY) && down_key_state == 0)
+            {
+                // increment selected
+                selected += 1;
+                // call functin to highlight button
+                highlight_down(buttons);
+                // this if statement is used to remove the hightlight after first run
+                if (count > 0)
+                {
+                    // remove the previous highlight
+                    remove_highlight_down(buttons);
+                }
+                down_key_state = 1;
+                count += 1;
+            }
+            else if (key_up(DOWN_KEY) && down_key_state == 1)
+            {
+                down_key_state = 0;
+            }
+            // Up key pressed
+            else if (key_down(UP_KEY) && up_key_state == 0)
+            {
+                // decrease seletced by 1
+                selected -= 1;
+                // call function to highlight button
+                highlight_up(buttons);
+                // this if statement is used to remove the hightlight after first run
+                if (count > 0)
+                {
+                    // remove the previous highlight
+                    remove_highlight_up(buttons);
+                }
+                up_key_state = 1;
+                count += 1;
+            }
+            else if (key_up(UP_KEY) && up_key_state == 1)
+            {
+                up_key_state = 0;
+            }
+        }
+
+        // Highlight button on down arrow key (up arrow key).
+        void highlight_down(vector<Button*> buttons)
```

Comment:
You could reduce code duplication by merging both highlight and remove_highlight functions, this would also simplify the class.
*Obviously depends on your future vision for the class*